internal/engine/wazevo/backend/isa/amd64/machine.go

```diff
@@ -2191,3 +2209,202 @@ func (m *machine) lowerFcvtFromUint(rn, rd operand, src64, dst64 bool) {
 	m.insert(endTarget)
 	m.copyTo(tmpXmm, rd.r)
 }
+
+func (m *machine) lowerVanyTrue(instr *ssa.Instruction) {
+	x := instr.Arg()
+	rm := m.getOperand_Reg(m.c.ValueDefinition(x))
+	rd := m.c.VRegOf(instr.Return())
+
+	cmp := m.allocateInstr()
+	cmp.asXmmCmpRmR(sseOpcodePtest, rm, rm.r)
+	m.insert(cmp)
+
+	setcc := m.allocateInstr()
+	setcc.asSetcc(condNZ, rd)
+	m.insert(setcc)
+
+	// Clear the irrelevant bits.
+	and := m.allocateInstr()
+	and.asAluRmiR(aluRmiROpcodeAnd, newOperandImm32(1), rd, false)
+	m.insert(and)
```

Comment:
this is not good from regalloc's perspective, as you are defining rd above this. what happens if this is spilled

Reply:
doing it (multiple define) for temporally regs is fine, but absolutely no for `rd`.

Reply:
IIRC I ended up doing this because most of instructions are defKindNone. But now that there is defineUninitializedReg I should be able to use that

Reply:
no you don't need defineUninitializedReg

Reply:
> most of these instructions are defKindNone

how do you think this is relevant in this comment ?

Reply:
no it was not relevant specifically to this comment 👍